slackbot-background/background.go

```diff
@@ -125,53 +123,87 @@ func InitConfig() {
 	globalConfig.allowedGroups = strings.Split(os.Getenv("ALLOWED_LDAP_GROUPS"), ",")
 }
 
-func FoxsecSlackBot(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(http.StatusOK)
-
-	err := verifySignature(r)
+func alertEscalator(ctx context.Context) error {
+	alerts, err := DB.GetAllAlerts(ctx)
```

Comment:
Will this function eventually become slow/problematic if we are fetching everything?

Also, could this potentially be executed concurrently? Does the alert update happen in a transaction if so?

Reply:
Yeah it will eventually become slow. One thought I had was to actually have an additional cleanup task that would remove alerts from datastore if they are older than X days (probably ~10) and their status != "NEW". Since they are stored in bigquery, no reason to keep the alerts in datastore after they have been acted upon except for debugging.

>Does the alert update happen in a transaction if so?

Yeah, the `common` lib/datastore client does all updates in transactions.

Reply:
👍 yeah this sounds good -- once we need it I imagine it would be simple enough to add since we already have similar scheduled code for whitelist cleanup.